### PR TITLE
Kids 1049 integrate qr & transaction

### PR DESCRIPTION
**Pull request by Daniela510**

### CODE REVIEW

File: lib/features/giving_flow/models/organisation.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    required this.name,
+    required this.goal,
+    this.logoLink,
+    this.thankYou,
+  });
+
+  const Organisation.empty()
+      : this(
+            collectGroupId: '',
+            name: 'Mock Organisation Long Name',
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/models/organisation.dart:15]: Will the user see these texts? And is that discussed with Andy?

File: lib/features/giving_flow/models/organisation.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  final String goal;
+  final String? logoLink;
+  final String? thankYou;
+
+  @override
+  List<Object?> get props => [collectGroupId, name, goal];
+
+  factory Organisation.fromMap(Map<String, dynamic> map) {
+    return Organisation(
+      collectGroupId: map['collectGroupId'] ?? 'no id fetched',
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/models/organisation.dart:34]: Same here

File: lib/features/giving_flow/repository/data_providers/create_transaction_bff_backend_data_provider.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+
+import 'package:givt_app_kids/features/giving_flow/models/transaction.dart';
+
+class CreateTransactionBffBackendDataProvider {
+  Future<void> createTransaction({required Transaction transaction}) async {
+    const hardcodedToken =
+        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiI0NkRBNTU5RC1CQ0M4LTQ0RjgtQjVFOC0wNzQwRjMxNEQzODMiLCJ1bmlxdWVfbmFtZSI6ImdpdnQtdGVhbUBnaXZ0YXBwLm5ldCIsInJvbGUiOiJnaXZ0T3BlcmF0b3IiLCJBcHBJZCI6IkludGVybmFsIGRhc2hib2FyZCAoUmV0b29sKSIsIm5iZiI6MTY4MTgxMDY4OSwiZXhwIjoxOTIxODEwNjg5LCJpYXQiOjE2ODE4MTA2ODksImlzcyI6Imh0dHBzOi8vYXBpLmdpdnRhcHAubmV0In0.y4QL_WZcPGwpmD471RmnMwFtMFlNYvzcpbiblqgHkBw';
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/repository/data_providers/create_transaction_bff_backend_data_provider.dart:11]: Hardcoded token?

File: lib/features/giving_flow/repository/data_providers/organisation_bff_data_provider.dart [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import '../../models/organisation.dart';
+
+class OrganisationBffDataProvider {
+  Future<Organisation> fetchOrganisationDetails(String? mediumId) async {
+    final url = Uri.https(
+      'dev-backend.givtapp.net',
+      'givt4kidsservice/v1/organisation/organisation-detail/$mediumId',
+    );
+    String authorizationToken =
+        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiI0NkRBNTU5RC1CQ0M4LTQ0RjgtQjVFOC0wNzQwRjMxNEQzODMiLCJ1bmlxdWVfbmFtZSI6ImdpdnQtdGVhbUBnaXZ0YXBwLm5ldCIsInJvbGUiOiJnaXZ0T3BlcmF0b3IiLCJBcHBJZCI6IkludGVybmFsIGRhc2hib2FyZCAoUmV0b29sKSIsIm5iZiI6MTY4MTgxMDY4OSwiZXhwIjoxOTIxODEwNjg5LCJpYXQiOjE2ODE4MTA2ODksImlzcyI6Imh0dHBzOi8vYXBpLmdpdnRhcHAubmV0In0.y4QL_WZcPGwpmD471RmnMwFtMFlNYvzcpbiblqgHkBw';
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/repository/data_providers/organisation_bff_data_provider.dart:15]: Again?

File: lib/features/giving_flow/screens/choose_amount_slider_screen.dart [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                    onPressed: state.amount == 0
+                        ? null
+                        : () async {
+                            if (state is CreateTransactionUploadingState) {
+                              return;
+                            }
+                            var transaction = Transaction(
+                              userId: _profilesCubit.state.activeProfile.id,
+                              collectGroupId:
+                                  '3fa85f64-5717-4562-b3fc-2c963f66afa6', // TODO use real value
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/screens/choose_amount_slider_screen.dart:200]: Does this need to be changed before merging?

File: lib/features/giving_flow/screens/choose_amount_slider_screen.dart [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+                              amount: state.amount,
+                            );
+
+                            context
+                                .read<CreateTransactionCubit>()
+                                .createTransaction(transaction: transaction);
+
+                            // AnalyticsHelper.logEvent(
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/screens/choose_amount_slider_screen.dart:208]: Don't we need to log to Amplitude?

File: lib/features/giving_flow/screens/success_screen.dart [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+              right: 40,
+            ),
+            child: ElevatedButton(
+              onPressed: () async {
+                Navigator.of(context).popUntil(
+                  ModalRoute.withName("/"),
+                );
+                //CLEAR ORGANISATION STATE
+                context.read<OrganisationCubit>().clearOrganisation();
+                // await AnalyticsHelper.logEvent(
```
Comment (MaikelStuivenberg) [inline lib/features/giving_flow/screens/success_screen.dart:97]: Same Amplitude question?

File: lib/features/giving_flow/models/organisation.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/giving_flow/models/organisation.dart:15]: No the user wont see this

File: lib/features/giving_flow/models/organisation.dart (hunk 2 above)
Comment (Daniela510) [reply lib/features/giving_flow/models/organisation.dart:34]: also wont be seen by the user

File: lib/features/giving_flow/screens/choose_amount_slider_screen.dart (hunk 5 above)
Comment (Daniela510) [reply lib/features/giving_flow/screens/choose_amount_slider_screen.dart]: Yep, I updated it

**Pull request closed without merging**